In [2]:
import torch
import numpy as np
import networkx as nx
import matplotlib.pyplot as plt

In [9]:
from torch_geometric.datasets import KarateClub

dataset = KarateClub()

print(f'Number of graphs: {len(dataset)}')
print(f'Number of features: {dataset.num_features}')
print(f'Number of classes: {dataset.num_classes}')

data = dataset[0]  # Get the first graph object.
print(data)

Number of graphs: 1
Number of features: 34
Number of classes: 4
Data(x=[34, 34], edge_index=[2, 156], y=[34], train_mask=[34])


The following result of the graph is expected: **Data(x=[34, 34], edge_index=[2, 156], y=[34], train_mask=[34])**

- x=[34, 34] is the node feature matrix with shape (number of nodes, number of features). In our case, it means that we have 34 nodes (our 34 members), each node being associated to a 34-dim feature vector. **Node Feature Matrix**

- edge_index=[2, 156] represents the graph connectivity. 156 edges with first row the source nodes and the second the target nodes.

- y=[34], representing the category label of every node

- train_mask = [34] is a boolean mask (list of True and False values) indicating which nodes should be used for training.
